plugin/goto.py

```diff
@@ -15,53 +15,72 @@
 class LspGotoCommand(LspTextCommand):
 
     method = ''
+    fallback_command = ''
 
-    def is_enabled(self, event: Optional[dict] = None, point: Optional[int] = None, side_by_side: bool = False) -> bool:
+    def is_enabled(
+        self,
+        event: Optional[dict] = None,
+        point: Optional[int] = None,
+        side_by_side: bool = False,
+        fallback: bool = False
+    ) -> bool:
         return super().is_enabled(event, point)
 
     def run(
         self,
         _: sublime.Edit,
         event: Optional[dict] = None,
         point: Optional[int] = None,
-        side_by_side: bool = False
+        side_by_side: bool = False,
+        fallback: bool = False
     ) -> None:
         session = self.best_session(self.capability)
         position = get_position(self.view, event, point)
         if session and position is not None:
```

Comment:
@rchl I was thininkig about one thing. What if we change this to something like
```python
if session and position is not None:
    # ...
elif position is not None:
    self._handle_no_results(True, side_by_side)
```
This way the `lsp_symbol_definition` with `"fallback": true` can be a drop-in replacement for the `goto_definition`(I mean we wouldn't need the context with the capatibility check) because if there is no capability(or no LSP at all) it would be just an equivalent for the `goto_definition`

Reply:
Technically it's not necessary and it's a bit safer to not use the LSP command when there are no active sessions.

Reply:
Hm, what if we do this only when the `fallback` is set to `true`
```python
if session and position is not None:
    # ...
else:
    self._handle_no_results(fallback, side_by_side)
```
This way, users who have `{"key": "lsp.session_with_capability", "operator": "equal", "operand": "definitionProvider"}` in their key binding notice a difference **only** if for some reasons `position is None`(they would see `No results found` in the status bar, which is perfectly fine I think).

But at the same, users who want to have one command(`lsp_symbol_definition`) everywhere can achieve that by having it with `"fallback": true` argument without any additional context(e.g. `{"keys": [<key>], "command": "lsp_symbol_definition", "args": { "fallback": true }`)

It can be useful for cases where there is no context support, like, if you want to go to the definition with CMD + mouse click. Currently there is no way to achieve this, because `*.sublime-mousemap` files don't support context, so you either use the built-in `goto_definition`(which is not always accurate) or `lsp_symbol_definition`(which is not always available)

Reply:
right, mouse maps are a good argument for doing that I guess

Reply:
Ok, here we go 5a970b492ce731a6821b9bc1509649f24ba23894

Please let me know what do you think?

I tested it and it works great.